twine/settings.py

```diff
@@ -338,6 +340,21 @@ def check_repository_url(self) -> None:
                 repository_url, utils.DEFAULT_REPOSITORY, utils.TEST_REPOSITORY
             )
 
+        if repository_url.startswith(
+            ("https://pypi.org/", "https://test.pypi.org/", "https://upload.pypi.org/")
+        ) and repository_url not in [utils.DEFAULT_REPOSITORY, utils.TEST_REPOSITORY]:
+            raise exceptions.InvalidPyPIUploadURL(
+                f"The configured repository {repository_url} is not a known "
+                f"PyPI repository.\n"
+                f"Did you mean {utils.DEFAULT_REPOSITORY} "
+                f"or {utils.TEST_REPOSITORY} ?\n"
+                f"Check your --repository-url value or "
+                f"modify the url in {self.config_file} "
+                f"(if you are using -r or --repository),\n"
```

Comment:
I don't think the parenthetical is necessary or helpful as written

Reply:
Using ``--repository-url https://test.pypi.org/legacy`` will also raise `InvalidPyPIUploadURL`. Probably folks who use ``--repository-url`` won't know what is `~/.pypirc`?